examples/cpp-simple/server.cc

```diff
@@ -41,7 +46,11 @@ void WatchUpdates(std::shared_ptr<agones::SDK> sdk) {
               << std::flush;
   });
 }
-
+void join_threads(std::thread health, std::thread watch) {
```

Comment:
Instead of a function, I think we should use an RAII object to join the threads. This should work with the current code and all of its return statements, and it will be more future proof because it will continue to work even if we add new return statements in the future.

How about something like this (untested):

```cc
class ThreadJoiner {
 public:
  explicit ThreadJoiner(std::thread t)
    : t_(std::move(t)) {}

  ~ThreadJoiner() {
    t_.join();
  }
 private:
  std::thread t_;
};
```
then in main() use it like

```cc
int main() {
  ...
  ThreadJoiner health(std::thread(DoHealth, sd));
  ThreadJoiner watch(std::thread(WatchUpdates, sdk));
```

Reply:
Fancy!

Reply:
I also was thinking about doing so, but some issues with raising exceptions from destructor this could cause. However let stick to what you suggest.